daprdocs/content/en/developing-applications/middleware.md

```diff
@@ -28,19 +28,42 @@ metadata:
 spec:
   httpPipeline:
     handlers:
-    - name: oauth2
-      type: middleware.http.oauth2
-    - name: uppercase
-      type: middleware.http.uppercase
+      - name: oauth2
+        type: middleware.http.oauth2
+      - name: uppercase
+        type: middleware.http.uppercase
 ```
 
 As with other building block components, middleware components are extensible and can be found in the [supported Middleware reference]({{< ref supported-middleware >}}) and in the [components-contrib repo](https://github.com/dapr/components-contrib/tree/master/middleware/http).
 
 {{< button page="supported-middleware" text="See all middleware components">}}
 
+## App Middleware (Preview)
+
+Http middleware are executed when invoking Dapr HTTP API directly. There are other types of middlewares that should be applied in the context of app protocol, they are called app middleware and they are also executed when making service-to-service invocations. Valid use cases are token validation in a zero-trust environment or request transformation for a specific app endpoint.
+
+The same set of middlewares that can be applied to http middleware can also be applied to app middleware but instead of using `httpPipeline` you should use `appHttpPipeline`.
```

Comment:
```suggestion
Any middleware component that can be applied to HTTP middleware can also be applied to service-to-service invocation calls as a middleware component using`appHttpPipeline` configuration. The example below adds the `uppercase` middleware component for all outbound calls from the application that this configuration is applied to.
```

Reply:
This is a fairly weak example. Could you provide a more compelling example with the existing middleware components?